compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -699,75 +699,280 @@ TR_OSRLiveRangeAnalysis::TR_OSRLiveRangeAnalysis(TR::OptimizationManager *manage
      _liveVars(NULL)
    {}
 
-int32_t TR_OSRLiveRangeAnalysis::perform()
+bool TR_OSRLiveRangeAnalysis::shouldPerformAnalysis()
    {
-   if (comp()->getOption(TR_EnableOSR))
+   if (!comp()->getOption(TR_EnableOSR))
       {
-      static const char *disableOSRLiveRangeAnalysis = feGetEnv("TR_DisableOSRLiveRangeAnalysis");
-      //if (comp()->getOption(TR_DisableOSRLiveRangeAnalysis)) // add to Options later
-      if (disableOSRLiveRangeAnalysis)
-         {
-         if (comp()->getOption(TR_TraceOSR))
-            traceMsg(comp(), "OSR is enabled but OSR live range analysis is not.\n");
-         return 0;
-         }
+      if (comp()->getOption(TR_TraceOSR))
+         traceMsg(comp(), "Should not perform OSRLiveRangeAnalysis -- OSR Option not enabled\n");
+      return false;
       }
-   else
+   else if (comp()->isPeekingMethod())
       {
       if (comp()->getOption(TR_TraceOSR))
-         traceMsg(comp(), "OSR Option not enabled -- returning from OSR live range analysis.\n");
-      return 0;
+         traceMsg(comp(), "Should not perform OSRLiveRangeAnalysis -- Not required because we are peeking\n");
+      return false;
+      }
+   else if (!comp()->supportsInduceOSR())
+      {
+      if (comp()->getOption(TR_TraceOSR))
+         traceMsg(comp(), "Should not perform OSRLiveRangeAnlysis -- OSR is not supported\n");
```

Comment:
OSR is not supported? don't you mean that we are not in a voluntary OSR mode or something like that?

Reply:
`supportsInduceOSR` will return false for a variety of modes that don't support OSR, such as DLT, with tracing for the reasoning behind it. I'll extend the message to make this clearer.

Reply:
Ok that would be great - thanks!